# Description:
#   name: diag name
#   sf_at: diag level where it becomes Safe Fault
#   lf_at: diag level where it becomes Latent Fault
#   spf_at: diag level where it becomes Single Point Fault
#   auto_recovery: Determines whether the system will automatically recover when it recovers from an error.
#
# Note:
# empty-value for sf_at, lf_at and spf_at is "none"
# default values are:
#   sf_at: "none"
#   lf_at: "warn"
#   spf_at: "error"
#   auto_recovery: "true"
---
/**:
  ros__parameters:
    required_modules:
      autonomous_driving:
        /autoware/control/autonomous_driving/node_alive_monitoring: default
        /autoware/control/autonomous_driving/performance_monitoring/lane_departure: default
        /autoware/control/control_command_gate/node_alive_monitoring: default
        /autoware/control/autonomous_emergency_braking/performance_monitoring/emergency_stop: { sf_at: "none", lf_at: "warn", spf_at: "error", auto_recovery: "false"}

        /autoware/localization/node_alive_monitoring: default
        # /autoware/localization/performance_monitoring/matching_score: { sf_at: "warn", lf_at: "none", spf_at: "none" }
        # /autoware/localization/performance_monitoring/localization_accuracy: default

        /autoware/map/node_alive_monitoring: default

        /autoware/perception/node_alive_monitoring: default

        /autoware/planning/node_alive_monitoring: default
        /autoware/planning/performance_monitoring/trajectory_validation: default

        # /autoware/sensing/node_alive_monitoring: default

        /autoware/system/node_alive_monitoring: default
        /autoware/system/emergency_stop_operation: default
        /autoware/system/service_log_checker: { sf_at: "warn", lf_at: "none", spf_at: "none" }
        /autoware/system/duplicated_node_checker: default
        # /autoware/system/resource_monitoring: { sf_at: "warn", lf_at: "error", spf_at: "none" }
        # /autoware/system/duplicated_node_checker: default

        /autoware/vehicle/node_alive_monitoring: default

      external_control:
        /autoware/control/control_command_gate/node_alive_monitoring: default
        /autoware/control/autonomous_emergency_braking/performance_monitoring/emergency_stop: { sf_at: "none", lf_at: "warn", spf_at: "error", auto_recovery: "false"}
        /autoware/control/external_control/external_command_selector/node_alive_monitoring: default

        /autoware/system/node_alive_monitoring: default
        /autoware/system/emergency_stop_operation: default
        /autoware/system/service_log_checker: { sf_at: "warn", lf_at: "none", spf_at: "none" }

        /autoware/vehicle/node_alive_monitoring: default
